### PR TITLE
Refresh onfido-ruby after onfido-openapi-spec update (d65af1f)

### PR DESCRIPTION
**Pull request by devops-automation-onfido**
Auto-generated PR with changes till commit onfido/onfido-openapi-spec@d65af1fbb54a42c1ec8c54be4aaa4c434046fd78 (included)

Additional changes:
  - None

### CODE REVIEW

File: lib/onfido/webhook_event_verifier.rb [hunk 1]
```diff
@@ -14,7 +14,7 @@ def read_payload(event_body, signature)
 
       raise(OnfidoInvalidSignatureError, "Invalid signature for webhook event") unless OpenSSL.secure_compare(signature, event_signature)
 
-      WebhookEvent.build_from_hash(JSON.parse(event_body)["payload"])
+      WebhookEvent.build_from_hash(JSON.parse(event_body))
```
Comment (RiccardoPetteruti-Onfido) [inline lib/onfido/webhook_event_verifier.rb:17]: This is already modified on openapi side but not re-triggered here. Manually updated.

File: .github/workflows/ruby.yml [hunk 2]
```diff
@@ -12,7 +12,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        ruby-version: ['2.4', '2.5', '2.6', '2.7', '3.0', '3.1']
+        ruby-version: ['2.7', '3.0', '3.1']
```
Comment (RiccardoPetteruti-Onfido) [inline .github/workflows/ruby.yml:15]: versions <2.7 are no more supported.

File: lib/onfido/api_client.rb [hunk 3]
```diff
@@ -0,0 +1,431 @@
...
+    # Check if the given MIME is a JSON MIME.
+    # JSON MIME examples:
+    #   application/json
+    #   application/json; charset=UTF8
+    #   APPLICATION/JSON
+    #   */*
+    # @param [String] mime MIME
+    # @return [Boolean] True if the MIME is application/json
+    def json_mime?(mime)
+      (mime == '*/*') || !!(mime =~ /^application\/(?:[^;]+\+)?json(?:$|;)/i)
```
Comment (RiccardoPetteruti-Onfido) [inline lib/onfido/api_client.rb:252]: updated as it was generating a high alert.

File: lib/onfido/api_client.rb [hunk 4]
```diff
@@ -0,0 +1,431 @@
...
+      end
+    end
+
+    # Sanitize filename by removing path.
+    # e.g. ../../sun.gif becomes sun.gif
+    #
+    # @param [String] filename the filename to be sanitized
+    # @return [String] the sanitized filename
+    def sanitize_filename(filename)
+        filename.strip!
```
Comment (RiccardoPetteruti-Onfido) [inline lib/onfido/api_client.rb:331]: updated as it was generating a high alert.

Comment (dvacca-onfido) [reply lib/onfido/api_client.rb]: I believe they're not doing the same thing. Previous code was removing `\` and `/` while [strip](https://apidock.com/ruby/String/strip) is just removing leading and trailing whitespace...

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,6 @@
+# Changelog
+
+## v 03th May 2024
```
Comment (dvacca-onfido) [inline CHANGELOG.md:3]: This line can be probably be removed for now

File: onfido.gemspec [hunk 6]
```diff
@@ -1,34 +1,43 @@
...
+Gem::Specification.new do |s|
+  s.name        = "onfido"
+  s.version     = Onfido::VERSION
+  s.platform    = Gem::Platform::RUBY
+  s.authors     = ["OpenAPI-Generator"]
+  s.email       = [""]
+  s.homepage    = "https://openapi-generator.tech"
+  s.summary     = "Onfido API v3.6 Ruby Gem"
+  s.description = "The Onfido API (v3.6)"
+  s.license     = "Unlicense"
```
Comment (dvacca-onfido) [inline onfido.gemspec:27]: We should customize these fields (in generator configuration or via templating) before the actual delivery.

File: lib/onfido/api_client.rb [hunk 7]
```diff
@@ -0,0 +1,431 @@
...
+      end
+    end
+
+    # Sanitize filename by removing path.
+    # e.g. ../../sun.gif becomes sun.gif
+    #
+    # @param [String] filename the filename to be sanitized
+    # @return [String] the sanitized filename
+    def sanitize_filename(filename)
+      filename.split(/[\/\\]/).last
```
Comment (RiccardoPetteruti-Onfido) [inline lib/onfido/api_client.rb:331]: manually updated to avoid high alert.

File: lib/onfido/api_client.rb [hunk 8]
```diff
@@ -249,7 +249,7 @@ def basic_auth(conn)
     # @param [String] mime MIME
     # @return [Boolean] True if the MIME is application/json
     def json_mime?(mime)
-      (mime == '*/*') || !(mime =~ /Application\/.*json(?!p)(;.*)?/i).nil?
+      (mime == '*/*') || !(mime =~ /^Application\/.*json(?!p)(;.*)?/i).nil?
```
Comment (RiccardoPetteruti-Onfido) [inline lib/onfido/api_client.rb:252]: manually updated to avoid high alert.

File: lib/onfido/api_client.rb (hunk 3 above)
Comment (RiccardoPetteruti-Onfido) [reply lib/onfido/api_client.rb]: updated on generator side.

File: lib/onfido/api_client.rb (hunk 4 above)
Comment (RiccardoPetteruti-Onfido) [reply lib/onfido/api_client.rb]: updated on generator side.

File: lib/onfido/api_client.rb (hunk 7 above)
Comment (RiccardoPetteruti-Onfido) [reply lib/onfido/api_client.rb:331]: updated on generator side.

File: CHANGELOG.md (hunk 5 above)
Comment (RiccardoPetteruti-Onfido) [reply CHANGELOG.md]: done

File: onfido.gemspec (hunk 6 above)
Comment (RiccardoPetteruti-Onfido) [reply onfido.gemspec:27]: done on generator side.

File: lib/onfido/api_client.rb (hunk 8 above)
Comment (RiccardoPetteruti-Onfido) [reply lib/onfido/api_client.rb:252]: updated on generator side.

**Review by dvacca-onfido — APPROVED** [review]

**Pull request merged**